xbmc/peripherals/bus/PeripheralBus.h

```diff
@@ -54,7 +54,7 @@ namespace PERIPHERALS
     /*!
      * @return True if this bus needs to be polled for changes, false if this bus performs updates via callbacks
      */
-    bool NeedsPolling(void) const { return m_bNeedsPolling; }
+    bool NeedsPolling(void) const { CSingleLock lock(m_critSection); return m_bNeedsPolling; }
```

Comment:
sorry I'm a bit fuzzy today and I'm mixing interlocked operations with CAS which isn't  the same thing. Yes you are right, for CAS operations where we need to know if we won the write it adds a direct method call and a loop over it.

What I've been refering to in this discussion are the interlocked gurantees which I've mistakenly called CAS. It gives us total ordering through memory barriers and protects against torn read/write and if I remember correctly should also make sure that there's no stale value in another core's cache line.

This gurantee is enough for the scenarios we have here but I think we both agree that we shouldn't juse willy nilly replace everything with std::atomic and claim we're lock free :smile:  They do make a lot of sense in most places we use boolean flags and are concerned about threading.

sorry for the confusion :smile:  Also I should mention that whenever I sidetrack into these nitpicky c++ details it's not really to force a change to the PR but rather to learn and or educate unless I explicitly state that the PR is broken :smile:  

*note* I was wrong about specifying the memory ordering during declaration, apprently it uses memory_order_seq_cst by default and if anything else is needed one of the explicit load/store compare* methods need to be called to specify the memory ordering